test/index.test.js

```diff
@@ -0,0 +1,107 @@
+const lolex = require('lolex');
+const moment = require('moment-timezone');
+const { setDayStartInterval } = require('../src');
+
+describe('setDayStartInterval', () => {
+  let clock, intervalFn;
+  beforeEach(() => {
+    clock = lolex.install({
+      // This is 8pm PDT sharp on 2019-09-17.
+      now: moment(1568775600000).tz('America/Los_Angeles').toDate()
+    });
+    intervalFn = jest.fn();
+  });
+
+  afterEach(() => {
+    clock.uninstall();
+  });
+
+  describe('firing', () => {
+    it('should fire at the start of the next day', () => {
+      setDayStartInterval(intervalFn, { tz: 'America/Los_Angeles' });
+      expect(intervalFn).not.toHaveBeenCalled();
+
+      const start = Date.now();
+      clock.runToLast();
+      const timeElapsed = Date.now() - start;
+
+      // 4 hours to reach midnight from 8pm.
+      expect(timeElapsed).toBe(moment.duration(4, 'hours').asMilliseconds());
+      expect(intervalFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should fire at the start of every day after that', () => {
+      setDayStartInterval(intervalFn, { tz: 'America/Los_Angeles' });
+
+      // Align to the start of the first day.
+      clock.runToLast();
+
+      // Now test 3 days in a row.
+      for (let i = 0; i < 3; i++) {
+        intervalFn.mockClear();
+
+        const start = Date.now();
+        clock.runToLast();
+        const timeElapsed = Date.now() - start;
+
+        expect(timeElapsed).toBe(moment.duration(1, 'days').asMilliseconds());
+        expect(intervalFn).toHaveBeenCalledTimes(1);
+      }
+    });
+
+    it('is time-zone aware', () => {
+      setDayStartInterval(intervalFn, { tz: 'America/New_York' });
+      expect(intervalFn).not.toHaveBeenCalled();
+
+      const start = Date.now();
+      clock.runToLast();
+      const timeElapsed = Date.now() - start;
+
+      // 8pm PDT is 11pm EDT, hence 1 hour to reach midnight.
+      expect(timeElapsed).toBe(moment.duration(1, 'hours').asMilliseconds());
+      expect(intervalFn).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('error handling', () => {
+    // TODO(jeff): Figure out how to prevent the uncaught exception below from failing Jest.
```

Comment:
Jest has a `expect(() => { ... }).toThrow()` syntax you might be able to leverage - unless Jest is doing magic (highly possible), you could start the interval in the closure with the initial assertion + run the clock, then after it's thrown run the rest of your assertions

Reply:
Ok the problem is [this Jest issue](https://github.com/facebook/jest/issues/5620). But reading that over makes me realize that testing this in Node doesn't make sense because in Node, the process is going to exit when this exception occurs—the interval won’t continue running even given how i’ve structured the code. It's only in the browser that the interval would continue running.

Reply:
Decided to remove this https://github.com/mixmaxhq/day-start-interval/pull/1/commits/73d2d1a3cbd546d8e83ae33f03bc843e9f3774c5